### PR TITLE
Add support for light icon version.

### LINKED ISSUE #392: Modify the build functionality to accept light icons
Now the project doesn't take care of light themes. I think it would be easy enough to modify the build process in order to add specific icons for light themes.

### PR DESCRIPTION
**Pull request by JimiC**
***Fixes #392***

**Changes proposed:**
* [x] Add
* [ ] Prepare
* [ ] Delete
* [ ] Fix

**Things I've done:**
* [x] My pull request fixes an issue, I referenced the issue.

### CODE REVIEW

File: src/build/iconGenerator.js [hunk 1]
```diff
@@ -113,9 +163,30 @@ function buildJsonStructure(iconsFolderBasePath) {
...
+  // dark theme
+  defaultSchema.iconDefinitions._file
+    .iconPath = iconsFolderBasePath + 'file.svg';
+  defaultSchema.iconDefinitions._folder
+    .iconPath = iconsFolderBasePath + 'folder.svg';
+  defaultSchema.iconDefinitions._folder_open
+    .iconPath = iconsFolderBasePath + 'folder_opened.svg';
+
+  // light theme
+  // default file and folder related icon paths if not set,
```
Comment (jens1o) [inline src/build/iconGenerator.js]: Use` /*[...]*/` instead for this long comment.

File: tests/iconGenerator.test.js [hunk 2]
```diff
@@ -179,10 +352,25 @@ describe('generating icons', function () {
...
+            expect(fileNames[extension]).toEqual(definition);
+          });
+        });
+    });
+
   it('ensures each supported file extension that is supported by language ids ' +
     'has a language id referencing its definiton',
     function () {
-      var languageIds = iconGenerator.buildJsonStructure().files.names.languageIds;
+      var languageIds = iconGenerator.buildJsonStructure().files.languageIds;
```
Comment (jens1o) [inline tests/iconGenerator.test.js]: why this?

Comment (JimiC) [inline tests/iconGenerator.test.js]: It has been changed in line 153.

File: src/build/iconGenerator.js (hunk 1 above)
Comment (JimiC) [inline src/build/iconGenerator.js]: I'm used in using single-line comment style inside functions/methods even if they are multi-lined.
I only use comment-block styling if they are file header (usually license) or above functions/methods as descreptive comments.
If @robertohuertasm wished to follow the comment-block style inside functions I have no problem switching to that.

Comment (JimiC) [inline src/build/iconGenerator.js]: For more references see:
http://www.hongkiat.com/blog/source-code-comment-styling-tips/
http://javascript.crockford.com/code.html

Comment (robertohuertasm) [inline src/build/iconGenerator.js]: I'm ok with this style. VSCode itself uses this kind of style when you select several lines so no problem with it. Is it ok for you @jens1o ?

Comment (jens1o) [inline src/build/iconGenerator.js]: Yeah, it's ok. I think it's the basic code style to use comment blocks.

File: tests/iconGenerator.test.js (hunk 2 above)
Comment (jens1o) [inline tests/iconGenerator.test.js]: and why?

Comment (jens1o) [inline tests/iconGenerator.test.js]: Ah, okey.

File: src/build/iconGenerator.js [hunk 3]
```diff
@@ -36,12 +36,19 @@ function buildJsonStructure(iconsFolderBasePath) {
...
         var names = old.names;
+        var light = old.light;
         var icon = current.icon;
+        var hasLightVersion = current.light;
         var iconFolderType = 'folder_type_' + icon;
+        var iconFolderLightType = 'folder_type_light_' + icon;
         var folderPath = iconsFolderBasePath + iconFolderType;
+        var folderLightPath = iconsFolderBasePath + iconFolderLightType;
         var openFolderPath = folderPath + '_opened';
+        var openFolderLightPath = openFolderPath + '_opened';
```
Comment (JimiC) [inline src/build/iconGenerator.js]: Damn. There is a bug here.

**Comment by JimiC** [comment]
@robertohuertasm Do you think it would be wise to add a case in code where if the `file_light`, `folder_light` and `folder_light_open` get assigned, the code populates the related properties with the dark theme definition when a `light` version is not provided?

File: src/build/iconGenerator.js (hunk 3 above)
Comment (JimiC) [inline src/build/iconGenerator.js]: Fixed.

**Comment by robertohuertasm** [comment]
@JimiC, sorry but I'm not really sure to understand what you say. You first say that the light properties get assigned but then you later say `when the light version is not provided`?

I guess you're asking if we should provide light versions no matter if they're assigned or not, is that right?

**Comment by JimiC** [comment]
> ___if___ the file_light, folder_light and folder_light_open get assigned

Emphases on `if`.
An with 
> when the light version is not provided

I mean for the other icons other than the default

I think I'll implement this guard just in case we decide to have other default icons for `light`. With the current implementation if we do so it will break the `light` icons.

**Comment by robertohuertasm** [comment]
Better to discuss this via Slack or something because I'm not following you on this.

**Pull request merged**